tests/orm/test_querybuilder.py

```diff
@@ -852,6 +852,59 @@ def test_empty_filters(self):
         qb = orm.QueryBuilder().append(orm.Data, filters={'or': [{}, {}]})
         assert qb.count() == count
 
+    from aiida.orm import AbstractCode, Computer, InstalledCode, PortableCode, QueryBuilder
+
+    def test_abstract_code_filtering(self):
+        """Test that querying for AbstractCode correctly returns all code instances.
+
+        This tests the fix for issue #6687, where QueryBuilder couldn't find codes
+        when looking for AbstractCode due to a node_type mismatch.
+        """
+        # Set up test environment
+        computer = orm.Computer(
+            label='test_computer_abstract',
+            hostname='localhost',
+            transport_type='core.local',
+            scheduler_type='core.direct',
+        ).store()
+
+        # Create test codes of different types
+        installed_code = orm.InstalledCode(
+            label='test_installed_abstract',
+            computer=computer,
+            filepath_executable='/bin/bash',
+        ).store()
+
+        portable_code = orm.PortableCode(
+            label='test_portable_abstract',
+            filepath_executable='/bin/bash',
+        ).store()
+
+        try:
+            # Test 1: AbstractCode query should find all code types
+            qb_abstract = orm.QueryBuilder().append(orm.AbstractCode)
+            abstract_results = qb_abstract.all(flat=True)
+            assert installed_code in abstract_results, 'InstalledCode not found with AbstractCode query'
```

Comment:
I don't think these asserts will work, since the objects returned by the QueryBuilder will be different from the original objects that you stored, even though they point to the same DB node. I think you should compare the `id` of the nodes, something like

```suggestion
            abstract_ids = [code.id for code in abstract_results]
            assert installed_code.id in abstract_ids, 'InstalledCode not found with AbstractCode query'
```